lib/job-iteration/iteration.rb

```diff
@@ -206,11 +183,17 @@ def build_enumerator(params, cursor:)
     def assert_valid_cursor!(cursor)
       return if serializable?(cursor)
 
-      raise CursorError.new(
-        "Cursor must be composed of objects capable of built-in (de)serialization: " \
-        "Strings, Integers, Floats, Arrays, Hashes, true, false, or nil.",
-        cursor: cursor,
-      )
+      Deprecation.warn(<<~DEPRECATION_MESSAGE.chomp)
+        Cursor must be composed of objects capable of built-in (de)serialization:
+          Strings, Integers, Floats, Arrays, Hashes, true, false, or nil.
+        #{self.class.name}#build_enumerator's Enumerator provided:
+          #{begin
+              cursor.inspect
+            rescue
+              Object.instance_method(:inspect).bind(cursor).call
+            end}
```

Comment:
Rubocop didn't like my inline `rescue`. 🤷 

This is so we can handle any cursor, including `BasicObject.new`. When that blows up, it won't be us. 😅

Reply:
If we believe showing the `cursor` as part of the error message is worth it. We could move this into a private method

```ruby
def cursor_to_s
  cursor.inspect
rescue
  Object.instance_method(:inspect).bind(cursor).call
end

Reply:
Being honest maybe just mentioning the class of the job that fails is enough for the developer to figure out where to look and what to fix.

If someone is using a custom cursor and gets this error message:

```
Cursor must be composed of objects capable of built-in (de)serialization:
  Strings, Integers, Floats, Arrays, Hashes, true, false, or nil.
#{self.class.name}#build_enumerator's Enumerator provided an invalid cursor object.
```

Could be more than enough